checks/dns.rb

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+# Check to verify forward / reverse dns matches hostname
+
+def error_exit(message, code)
+  $stderr.puts message
+  exit code
+end
+
+@hostname = `hostname -f`.chomp
+@ip = `getent ahosts #{@hostname} | awk '/STREAM/ { print $1 }'`.chomp
+@reverse = `getent hosts #{@ip} | awk '{ print $2 }'`.chomp
+
+FORWARD = "Unable to resolve forward DNS for #{@hostname}"
+
+POINTS = "Forward DNS points to #{@ip} which is not configured on this server"
+
+REVERSE = "Reverse DNS #{@reverse} does not match hostname #{@hostname}"
+
+if @ip.empty?
+  error_exit(FORWARD, 2)
+end
+
+`ip a | grep #{@ip} | awk '{print $2 }' | rev | cut -c 4- | rev`.chomp
```

Comment:
Is the cut intended to strip off the netmask? In that case I'd recommend to let cut split on the slash. You can also filter in awk:
```bash
ip -o a | awk '/ #{@ip}\// { print $4 }' | cut -d/ -f1
```

Note I used ```ip -o``` which is the programmatic output. I also modified the search slightly so that it has boundries. This prevents matching ```10.0.0.10``` if you're looking for ```10.0.0.1```.